src/main/java/roomescape/auth/CheckAdminInterceptor.java

```diff
@@ -0,0 +1,37 @@
+package roomescape.auth;
+
+import jakarta.servlet.http.Cookie;
+import jakarta.servlet.http.HttpServletRequest;
+import jakarta.servlet.http.HttpServletResponse;
+import org.springframework.web.servlet.HandlerInterceptor;
+import roomescape.member.Member;
+import roomescape.member.MemberService;
+
+public class CheckAdminInterceptor implements HandlerInterceptor {
+
+    private final MemberService memberService;
+    private final JwtTokenProvider jwtTokenProvider;
+    private final CookieValueExtractor cookieValueExtractor;
+
+    public CheckAdminInterceptor(MemberService memberService, JwtTokenProvider jwtTokenProvider,
+                                 CookieValueExtractor cookieValueExtractor) {
+        this.memberService = memberService;
+        this.jwtTokenProvider = jwtTokenProvider;
+        this.cookieValueExtractor = cookieValueExtractor;
+    }
+
+    @Override
+    public boolean preHandle(HttpServletRequest request, HttpServletResponse response, Object handler) throws Exception {
+        Cookie[] cookies = request.getCookies();
+        String token = cookieValueExtractor.extractToken(cookies);
+        Long memberId = jwtTokenProvider.getMemberIdByToken(token);
+        Member member = memberService.getMemberById(memberId);
+
+        if (member == null || !member.getRole().equals("ADMIN")) {
```

Comment:
role 을 꺼내와서 검증하는 것보단, member에서 권한을 확인하는 로직을 만들면 어떨까요?
```suggestion
        if (member == null || !member.isAdmin()) {
```

Reply:
필드 값 확인, 검증 같은 로직은 객체가 직접 확인하는 게 좀 더 Member 객체의 역할에 맞다는 생각이 드네요! 객체의 필드값을 직접 꺼내와서 확인하는 로직은 가독성도 안좋고, 추후 다른 코드에서 검증이 더 필요할 때 재사용성도 좋지 않아 보입니다. 수정하였습니다!